pkg/globalnet/controllers/global_ingressip_controller.go

```diff
@@ -148,12 +177,56 @@ func (c *globalIngressIPController) onDelete(ingressIP *submarinerv1.GlobalIngre
 
 	key, _ := cache.MetaNamespaceKeyFunc(ingressIP)
 
+	if ingressIP.Spec.ServiceRef != nil {
+		chainName := ingressIP.GetAnnotations()[kubeProxyIPTableChainAnnotation]
+		if chainName == "" {
+			klog.Warningf("%q annotation is missing on %q", kubeProxyIPTableChainAnnotation, key)
+			return nil, true
+		}
+
+		err := c.iptIface.RemoveIngressRulesForService(ingressIP.Status.AllocatedIP, chainName)
```

Comment:
`RemoveIngressRulesForService` returns an error if the passed global IP is empty. To avoid an extraneous error logged, we can either check for empty `ingressIP.Status.AllocatedIP` on L180 or ignore an empty IP in `RemoveIngressRulesForService`

Reply:
I'm checking the globalIP and Chainname in RemoveIngressRulesForService API. Instead of ignoring the empty, lets log an error (ideally this should not happen, but will help us to know if there is some corner case)

Reply:
It could happen if the `GlobalIngressIP` gets deleted prior to allocating the IP or if IP allocation failed. But that should be uncommon so logging an error is fine.